src/io.rs

```diff
@@ -45,4 +46,8 @@ pub fn append_to_file(filename: PathBuf, content: String) -> Result<usize, io::E
         .open(&filename)?;
 
     file.write(content.as_bytes())
+}
+
+pub fn read_file(filename: &str) -> Result<usize, Error> {
+    return Ok(1)
```

Comment:
I'll actually push my refactoring of that soon, that has the functions de-tangled and actually implemented.